kymatio/backend/tensorflow_backend.py

```diff
@@ -6,12 +6,26 @@ class TensorFlowBackend(NumpyBackend):
     name = 'tensorflow'
 
     @staticmethod
-    def concatenate(arrays):
-        return tf.stack(arrays, axis=1)
+    def concatenate(arrays, dim=1):
+        return tf.stack(arrays, axis=dim)
 
     @staticmethod
     def modulus(x):
         norm = tf.abs(x)
 
         return norm
 
+    @staticmethod
+    def reshape_input(x, signal_shape, n_inserted_dims=0):
```

Comment:
Not following this `n_inserted_dims`, `n_kept_dims` business. Seems like the former is always `1` and the second is sometimes `1`, sometimes `2`. Should the latter case be handled by a `stack`–`concatenate` distinction?

Reply:
i have raised an issue https://github.com/kymatio/kymatio/issues/895

> Should the latter case be handled by a stack–concatenate distinction?

I did this to avoid introducing two kymatio primitives (stack and concatenate) when `reshape_output` does the job up to now. Although i agree that "stack" and "concatenate" are more recognizable in terms of what they do.
I am happy to make that distinction in the backends if you think it's helpful to understand the logic here.

Reply:
> The former is always one for 1D. It would be two for 2D and three for 3D. Since i'm implementing this in a dimension-agnostic way, i need to pass `n_inserted_dims` as an input argument.

We're adding two singleton dimensions in 2D and three in 3D? Are you sure? If I understand correctly, this added dimension corresponds to the channel and is one for all the dimensions.

> For me, the extra `1` in `out_type="dict"` has no justification and we should present dict values as `(batch_size, time)`, just like with `"list"`.

I agree. We can discuss further on the issue.

Reply:
> We're adding two singleton dimensions in 2D and three in 3D? Are you sure? 

My bad. We're adding one singleton dimension no matter what the input dimension is.
2D code

```python
        batch_shape = input.shape[:-2]
        signal_shape = input.shape[-2:]
        input = input.reshape((-1,) + signal_shape)
        S = scattering( ...
        scattering_shape = S.shape[-3:]
        new_shape = batch_shape + scattering_shape
        S = S.reshape(new_shape)
```


So, `n_inserted_dims` doesn't need to be modifiable. I'll update accordingly.

Reply:
fixed in 4a274be